src/core/server/saved_objects/import/create_saved_objects.ts

```diff
@@ -130,6 +132,22 @@ export const createSavedObjects = async <T>({
             });
           }
 
+          // Some visualization types will need special modifications, like TSVB visualizations
+          const timelineExpression = extractTimelineExpression(object);
+          if (!!timelineExpression && !!dataSourceTitle) {
+            // Get the timeline expression with the updated data source name
+            const modifiedExpression = updateDataSourceNameInTimeline(
+              timelineExpression,
+              dataSourceTitle
+            );
+
+            // @ts-expect-error
+            const timelineStateObject = JSON.parse(object.attributes?.visState);
+            timelineStateObject.params.expression = modifiedExpression;
+            // @ts-expect-error
+            object.attributes.visState = JSON.stringify(timelineStateObject);
+          }
+
```

Comment:
Non-blocking: this code is duplicate with the change in `check_conflict_for_data_source.ts`. Can we extract to a file or a function that we can reuse? If not able to complete by this release, can we create an issue for tracking?

Reply:
Sure will create an issue. THanks zhongnan